src/sentry/api/serializers/models/team.py

```diff
@@ -178,8 +178,24 @@ class TeamSerializerResponse(_TeamSerializerResponseOptional):
     avatar: SerializedAvatarFields
 
 
+# We require a third Team Response TypedDict that inherits like so:
+# TeamSerializerResponse
+#   * BaseTeamSerializerResponse
+#   * _TeamSerializerResponseOptional
+# instead of having this inheritance:
+# BaseTeamSerializerResponse
+#   * _TeamSerializerResponseOptional
+# b/c of how drf-spectacular builds schema using @extend_schema. When specifying a DRF serializer
+# as a response, the schema will include all optional fields even if the response body for that
+# request never includes those fields. There is no way to have a single serializer that we can
+# manipulate to exclude optional fields at will, so we need two separate serializers where one
+# returns the base response fields, and the other returns the combined base+optional response fields
```

Comment:
does `_TeamSerializerResponseOptional` exist? or is this just a placeholder to allow for optional fields in `TeamSerializerResponse`?

Reply:
Yes it exists [here](https://github.com/getsentry/sentry/blob/130701e13ec109cea61f609aa6cfab83f46f0936/src/sentry/api/serializers/models/team.py#L159). We put optional fields in there that may not be set